emu/src/arm7tdmi.rs

```diff
@@ -122,6 +122,64 @@ impl Arm7tdmi {
         }
     }
 
+    fn data_transfer_immediate_offset(&mut self, op_code: ArmModeOpcode) -> bool {
+        let pre_post = op_code.get_bit(24);
+        let up_down = op_code.get_bit(23);
+        let _write_back = op_code.get_bit(21);
+        let load_store = op_code.get_bit(19);
+        let rn_base_register = op_code.get_bits(16..=19);
+        let rd_source_destination_register = op_code.get_bits(12..=15);
+        let transfer_type = HalfwordTransferType::from(op_code.get_bits(5..=6) as u8);
+        let immediate_offset_high = op_code.get_bits(8..=11);
+        let immediate_offset_low = op_code.get_bits(0..=3);
+
+        let offset = (immediate_offset_high << 4) | immediate_offset_low;
+        let mut address = rn_base_register;
+
+        if rn_base_register == 0xF {
+            // prefetching
+            let v: u32 = self.registers.program_counter().try_into().unwrap();
+            address = address.wrapping_add(v + 8);
+        }
+
+        if pre_post {
+            if up_down {
+                address = address.wrapping_add(offset);
+            } else {
+                address = address.wrapping_sub(offset);
+            }
+        }
+
+        if load_store {
+            todo!("load from mem")
+        } else {
+            let value = if rd_source_destination_register == 0xF {
+                let pc: u32 = self.registers.program_counter().try_into().unwrap();
+                pc + 12
+            } else {
+                self.registers
+                    .register_at(rd_source_destination_register as usize)
+            };
+
+            match transfer_type {
+                HalfwordTransferType::UnsignedHalfwords => {
+                    if let Ok(mut mem) = self.memory.lock() {
+                        mem.write_at(address, value.get_bits(0..=7) as u8);
+                        mem.write_at(address + 1, value.get_bits(8..=15) as u8);
+                    }
+                }
+                _ => unreachable!("HS flags can't be != from 01 for LOAD (L)=0"),
```

Comment:
```suggestion
                _ => unreachable!("HS flags can't be != from 01 for STORE (L=0)"),
```

Reply:
Eheheh easy to confuse with all those flags